packages/hdwallet-keepkey/src/transport.ts

```diff
@@ -92,7 +92,13 @@ export class Transport extends core.Transport {
     const buffer = new Uint8Array(9 + 2 + msgLength);
     buffer.set(first.slice(0, Math.min(first.length, buffer.length)));
 
-    for (let offset = first.length; offset < buffer.length; ) {
+    // msgLength includes the preamble overhead "?" for each chunk.
+    // We don't want the preamble from each chunk in the buffer (though, curiously, the preamble
+    // for the first chunk is included in the buffer for some reason).
+    // We do, however, have to count the preamble in each chunk to make the proper number
+    // of chunked reads
+    let msgLenReceived = 64; // already have first buffer
+    for (let offset = first.length; msgLenReceived < msgLength; msgLenReceived += 64) {
```

Comment:
weird ordering of the for loop vars
```suggestion
    let offset = first.length;
    for (let msgLenReceived = 64; msgLenReceived < msgLength; msgLenReceived += 64) {
```